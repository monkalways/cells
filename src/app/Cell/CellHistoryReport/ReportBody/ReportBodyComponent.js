import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Grid, Typography } from '@material-ui/core';
import { format } from 'date-fns';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  report: PropTypes.shape({
    cellName: PropTypes.string.isRequired,
    userLabel: PropTypes.string.isRequired,
    startTime: PropTypes.instanceOf(Date).isRequired,
    endTime: PropTypes.instanceOf(Date).isRequired,
    lastOccupantName: PropTypes.string,
  }).isRequired,
};

const dateTimeFormat = 'yyyy/MM/dd HH:mm';

const ReportBodyComponent = ({ classes, report }) => (
  <Grid item className={classes.container} xs={12}>
    <Typography variant="body1" className={classes.boldText}>
      Edmonton Police Service
    </Typography>
    <Typography variant="body1" className={classes.row}>
      <span className={classes.italicText}>Printed: </span>
      {format(new Date(), dateTimeFormat)} by {report.userLabel}
    </Typography>
    <Typography variant="body1" className={classes.row}>
      <span className={classes.boldText}>Cell ID:</span>
      <span>{report.cellName}</span>
    </Typography>
    <Typography variant="body1" className={classes.row}>
      <span className={classes.boldText}>Reporting Period:</span>
      <span>
        {format(report.startTime, dateTimeFormat)} to{' '}
        {format(report.endTime, dateTimeFormat)}
      </span>
    </Typography>
    <Typography variant="body1" className={classes.row}>
      <span className={classes.boldText}>Last Occupant during period:</span>
      <span>{report.lastOccupantName || 'N/A'}</span>
    </Typography>
  </Grid>
);

ReportBodyComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  container: {
    margin: theme.spacing.unit,
    height: theme.spacing.unit * 26,
    boxShadow: '2px 2px 2px 2px rgba(0, 0, 0, 0.2)',
  },
  boldText: {
    fontWeight: 500,
    marginRight: theme.spacing.unit * 2,
  },
  italicText: {
    fontStyle: 'italic',
    marginRight: theme.spacing.unit * 2,
  },
  row: {
    marginBottom: theme.spacing.unit * 3,
  },
}))(ReportBodyComponent);
