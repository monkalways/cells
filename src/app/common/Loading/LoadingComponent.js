import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
