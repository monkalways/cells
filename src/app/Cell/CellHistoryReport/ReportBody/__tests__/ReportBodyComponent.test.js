import React from 'react';
import { shallow } from 'enzyme';
import { ReportBodyComponent } from '../ReportBodyComponent';
import Loading from '../../../../common/Loading';

describe('ReportBodyComponent', () => {
  let isLoadingReport;

  beforeEach(() => {
    isLoadingReport = false;
  });

  const setup = () => {
    const classes = {};
    return shallow(<ReportBodyComponent
      classes={classes}
      report={{
        cellName: 'c1',
        userLabel: 'Doe. J',
        lastOccupantName: 'Test User 1',
        startTime: new Date(Date.UTC(2019, 2, 11, 0)),
        endTime: new Date(Date.UTC(2019, 2, 12, 0)),
        movementHistories: [
          {
            time: new Date(Date.UTC(2019, 2, 11, 0)),
            detaineeName: 'Test User 2',
            sourceCellName: 'c1',
            detinationCellName: 'c2',
          },
        ],
      }}
      isLoadingReport={isLoadingReport}
    />);
  };

  it('should render Loading component if isLoadingReport is true', () => {
    isLoadingReport = true;

    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Loading)).toExist();
  });

  it('should render cell history report if isLoadingReport is false', () => {
    isLoadingReport = false;

    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
