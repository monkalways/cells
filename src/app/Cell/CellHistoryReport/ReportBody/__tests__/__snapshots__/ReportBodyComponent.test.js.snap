// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportBodyComponent should render Loading component if isLoadingReport is true 1`] = `
<WithStyles(Grid)
  alignContent="flex-start"
  container={true}
  justify="center"
  spacing={8}
>
  <WithStyles(LoadingComponent) />
</WithStyles(Grid)>
`;

exports[`ReportBodyComponent should render cell history report if isLoadingReport is false 1`] = `
<WithStyles(Grid)
  alignContent="flex-start"
  container={true}
  justify="center"
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
    sm={12}
  >
    <WithStyles(Typography)
      variant="h5"
    >
      Cell History Report - c1
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Typography)
      variant="body1"
    >
      Edmonton Police Service
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="body1"
    >
      <span>
        Printed: 
      </span>
      2019/03/11 18:00
       by 
      Doe. J
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="body1"
    >
      <span>
        Cell ID:
      </span>
      <span>
        c1
      </span>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="body1"
    >
      <span>
        Reporting Period:
      </span>
      <span>
        2019/03/10 18:00
         to
         
        2019/03/11 18:00
      </span>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="body1"
    >
      <span>
        Last Occupant during period:
      </span>
      <span>
        Test User 1
      </span>
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(MovementHistoriesComponent)
    movementHistories={
      Array [
        Object {
          "detaineeName": "Test User 2",
          "detinationCellName": "c2",
          "sourceCellName": "c1",
          "time": 2019-03-11T00:00:00.000Z,
        },
      ]
    }
  />
</WithStyles(Grid)>
`;
